hs_core/hydroshare/users.py

```diff
@@ -338,7 +336,7 @@ def get_resource_list(creator=None, group=None, user=None, owner=None, from_date
         q.append(query)
 
     if published:
-        q.append(Q(doi__isnull=False))
+        q.append(Q(raccess__published=True))
```

Comment:
Here is the piece to fix filtering by published.   The rest of the changes are just optimizations and cleanup.